modules/common/src/main/scala/com/snowplowanalytics/snowplow/postgres/resources.scala

```diff
@@ -22,47 +22,51 @@ import doobie.util.ExecutionContexts
 import doobie.util.log.LogHandler
 import doobie.util.transactor.Transactor
 
-import fs2.concurrent.Queue
-
 import io.circe.Json
 
 import com.snowplowanalytics.iglu.client.Client
 
 import com.snowplowanalytics.snowplow.postgres.api.State
-import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig
-import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig.JdbcUri
-import com.snowplowanalytics.snowplow.postgres.streaming.source.BadData
+import com.snowplowanalytics.snowplow.postgres.config.DBConfig
+import com.snowplowanalytics.snowplow.postgres.config.DBConfig.JdbcUri
 
 object resources {
 
+  val FixedThreadPoolSize: Int = 32
+
   /** Initialise Blocking Thread Pool, Connection Pool, DB state and bad queue resources */
-  def initialize[F[_]: Concurrent: Clock: ContextShift](postgres: LoaderConfig,
+  def initialize[F[_]: Concurrent: Clock: ContextShift](postgres: DBConfig,
                                                         logger: LogHandler,
                                                         iglu: Client[F, Json]) =
     for {
       blocker <- Blocker[F]
-      badQueue <- Resource.liftF(Queue.bounded[F, BadData](128))
       xa <- resources.getTransactor[F](postgres.getJdbc, postgres.username, postgres.password, blocker)
-      keysF = for {
-        ci <- storage.query.getComments(postgres.schema, logger).transact(xa).map(_.separate)
-        (issues, comments) = ci
-        _ <- issues.traverse_(issue => Sync[F].delay(println(issue)))
-      } yield comments
-      keys <- Resource.liftF(keysF)
-      initState = State.init[F](keys, iglu.resolver).value.flatMap {
+      state <- Resource.liftF(initializeState(postgres, logger, iglu, xa))
+    } yield (blocker, xa, state)
+
+  def initializeState[F[_]: Concurrent: Clock](postgres: DBConfig,
+                            logger: LogHandler,
+                            iglu: Client[F, Json],
+                            xa: Transactor[F]): F[State[F]] = {
+    for {
+      ci <- storage.query.getComments(postgres.schema, logger).transact(xa).map(_.separate)
+      (issues, comments) = ci
+      _ <- issues.traverse_(issue => Sync[F].delay(println(issue)))
+      initState = State.init[F](comments, iglu.resolver).value.flatMap {
         case Left(error) =>
           val exception = new RuntimeException(s"Couldn't initalise the state $error")
           Sync[F].raiseError[State[F]](exception)
         case Right(state) =>
           Sync[F].pure(state)
       }
-      state <- Resource.liftF(initState)
-    } yield (blocker, xa, state, badQueue)
+      state <- initState
+    } yield state
+  }
 
   /** Get a HikariCP transactor */
   def getTransactor[F[_]: Async: ContextShift](jdbcUri: JdbcUri, user: String, password: String, be: Blocker): Resource[F, HikariTransactor[F]] =
     for {
-      ce <- ExecutionContexts.fixedThreadPool[F](32)
+      ce <- ExecutionContexts.fixedThreadPool[F](FixedThreadPoolSize)
```

Comment:
I will take a look at making this configurable - it's a good idea.

But..... 32 might be a sensible number even on nano/micro because it is not directly linked to number of processors, but rather to the number of threads awaiting connections to the database.  There are [details here](https://tpolecat.github.io/doobie/docs/14-Managing-Connections.html#about-threading).  In particular these sentences:

> An ExecutionContext for awaiting connection to the database. Because there can be an unbounded number of connections awaiting database access this should be a bounded pool.

> Also note that the number of JDBC connections is usually limited by the underlying JDBC pool. You may therefore want to limit your connection pool to the same size as the underlying JDBC pool as any additional threads are guaranteed to be blocked.

Reply:
True. The problem here is that the number of threads at spike load will result in constant thread switching leading to slow processing that might in turn make the app unresponsive to level of being killed. 
But this is likely a discussion for production-proofing when we have large-scale deployments.

Reply:
`availableProcessors` is definitely not the way to go by default. However since JDK9 JVM has means of working with that and the underlying instance features show not leak into container . Which means we probably want to move to the most recent LTS image for that reason.